File: ExampleApp/app/src/main/java/com/motiv/example/UserActivityPresenter.java
package com.motiv.example;

import com.motiv.example.dao.DaoRepository;
import com.motiv.example.dao.LocalStorage;

public class UserActivityPresenter implements UserActivityContract.Presenter {

    private UserActivityContract.View view;
    private GoApi goApi;
    private AuthApi authApi;
    private DaoRepository daoRepository;
    private LocalStorage localStorage;

    public UserActivityPresenter(
            UserActivityContract.View view,
            GoApi goApi,
            AuthApi authApi,
            DaoRepository daoRepository,
            LocalStorage localStorage) {
        this.view = view;
        this.goApi = goApi;
        this.authApi = authApi;
        this.daoRepository = daoRepository;
        this.localStorage = localStorage;
    }
}

File: ExampleApp/app/src/main/java/com/motiv/example/UserActivityPresenter.kt
 
package com.motiv.example
import com.motiv.example.dao.DaoRepository
import com.motiv.example.dao.LocalStorage

public class UserActivityPresenter(val view: UserActivityContract.View, val goApi: GoApi, val authApi: AuthApi, val daoRepository: DaoRepository, val localStorage: LocalStorage) : UserActivityContract.Presenter
